### PR TITLE
Fix completion record for labeled statement

### PR DESCRIPTION
**Pull request by addaleax**
`{labelAlthoughIMeantItToBeAKey: value}` strikes again :)

| Q                        | A 
| ------------------------ | ---
| Fixed Issues?            |  
| Patch: Bug Fix?          | yes
| Major: Breaking Change?  |
| Minor: New Feature?      |
| Tests Added + Pass?      | (where are the tests for getCompletionRecords?)
| Documentation PR Link    | 
| Any Dependency Changes?  |
| License                  | MIT

### CODE REVIEW

**Comment by nicolo-ribaudo** [comment]
Related - https://github.com/babel/babel/pull/13030. Tt conflicts, but it looks like it will be easy to fix the conflict since modified line would be https://github.com/babel/babel/pull/13030/files#diff-6828b1e955c2a5ac30a60be81472c92a36c5b90605ba94a4ebfcaedbbc41b43dR213

> where are the tests for getCompletionRecords

I think we only test it indirectly with `do` expressions, but if you can also add a new test in https://github.com/babel/babel/blob/main/packages/babel-traverse/test/family.js that would be appreciated!

**Comment by addaleax** [comment]
@nicolo-ribaudo Ah, thanks :) This isn’t urgent for me, I think it would be fine to wait until the other PR is merged.

**Comment by hzoo** [comment]
Since we don't have any getCompletionRecords tests, we could also add the tests for do expressions as mentioned https://github.com/babel/babel/tree/main/packages/babel-plugin-proposal-do-expressions/test/fixtures/do-expressions as a substitute?

```js
expect(do {
  a: 1
}).toBe(1);

expect(do {
  a: if (true) {
    "bar";
  }
}).toBe("bar");
```

**Comment by nicolo-ribaudo** [comment]
@addaleax The other PR has been merged!

**Comment by addaleax** [comment]
@nicolo-ribaudo Thank you, should be working now :)

**Review by JLHwung — APPROVED** [review]
LGTM

**Review by hzoo — APPROVED** [review]
awesome!

**Pull request merged**
